pennylane/tape/qscript.py

```diff
@@ -729,10 +729,20 @@ def set_parameters(self, params, trainable_only=True):
         if len(params) != required_length:
             raise ValueError("Number of provided parameters does not match.")
 
+        op_data = []
+        for pinfo in self._par_info:
```

Comment:
The purpose of the loop is to map each entry of `_par_info` to a list containing the corresponding operator's data.

This isn't needed in the old version because in (what used to be) line 734, `op.data` is modified in-place to store the new parameters. Since tuples aren't mutable, this isn't possible anymore. So we instead create a list containing the operator data, modify _that_ in place, cast it to a tuple, then set `op.data` to be the new tuple.

Reply:
ahh, I see the idea here. it wasn't clicking at first but I think this is great! might take the occasional performance hit when `trainable_only=True` and most parameters are not trainable, but this shouldn't be a problem

Reply:
Seeing that we're looking to deprecate `qscript.set_parameters` soon™, I wouldn't call it a problem either.